prime-router/docs/design/proposals/mapping-inventory-test-plan.md

```diff
@@ -0,0 +1,38 @@
+# Mapping Inventory Test Plan
+
+## Problem Statement
+As a ReportStream engineer, I want tests that will test the following statement: "The Universal Pipeline (UP) shall be
+able to transform valid NIST 2.5.1 messages losslessly to FHIR using the official HL7v2 mapping inventory as a guide. 
+UP shall be able to transform FHIR ELR messages to valid NIST 2.5.1."
+
+Currently, we have integration tests that test an unknown combination of parts of sender transforms, receiver 
+transforms, and mapping inventory rules. We want a clear and definitive way of knowing what is already tested, and we 
+want the mapping inventory testing to be thorough. This means testing not only that each piece of the message ends up 
+in the Primary Target (as defined in column J of the mapping inventory spreadsheet), but also that the conditions set
+forth in the Computable ANTLR column are tested. This will just test moving from HL7 V2 -> FHIR. We also need to test 
+that the resulting FHIR message can be converted back to HL7 V2 losslessly. To achieve this, we can reverse the order 
+of the files and ensure that the output from processing the FHIR file matches the HL7 V2 file. This will help keep the 
+number of files needed down. 
+
+Integration tests are currently rather challenging to read. At a minimum, we need a standard for naming the files. When
+testing mapping inventory segments, we will name files with the format `messageType_segment_#`. An example, 
+`ORU_R01_PID_1`. The corresponding FHIR file will be named the same, but will have a `.fhir` extension. This way, it is 
```

Comment:
Let's just double check that we need to distinguish based on messageType. Last time I looked at the mapping inventory, OML and ORU both pointed to the same PID, for example (I think it was PID, could have been a different one). If we do need to break these up by messageType, then we need to consider how we handle segments that are shared by messageTypes.

Reply:
That is fair. My assumption was that they would all live in ORU_R01 and then any once that deviated from it would be marked with their message type. Perhaps that is confusing though. I will rewrite this part.